core/src/main/java/org/wso2/testgrid/core/TestPlanExecutor.java

```diff
@@ -136,23 +143,89 @@ private void runScenarioTests(TestPlan testPlan, DeploymentCreationResult deploy
                 }
             }
         }
+
+        Set<String> appliedSenarios = new HashSet<String>();
+        if (testPlan.getScenarioConfig().getConfigChangeSets() != null) {
```

Comment:
Please assign "testPlan.getScenarioConfig().getConfigChangeSets()" to a variable and use it.